src/Microsoft.OData.CodeGen/Templates/ODataT4CodeGenerator.cs

```diff
@@ -2908,6 +2919,166 @@ internal void SetPropertyIdentifierMappingsIfNameConflicts(string typeName, IEdm
+
+    /// <summary>
+    /// CSharp code provider instance
+    /// </summary>
+    private static readonly CodeDomProvider CSharpProvider = new Microsoft.CSharp.CSharpCodeProvider();
+
+    /// <summary>
+    /// VB code provider instance
+    /// </summary>
+    private static readonly CodeDomProvider VBProvider = new Microsoft.VisualBasic.VBCodeProvider();
+
+    /// <summary>
+    /// Map invalid identifiers to valid names to avoid generating code that does not compile
+    /// </summary>
+    /// <param name="structuredType">Object containing properties to map</param>
+    internal void SetPropertyIdentifierMappingsIfInvalidIdentifier(IEdmStructuredType structuredType)
+    {
+        bool isLanguageCaseSensitive = this.context.TargetLanguage == LanguageOption.CSharp;
+        UniqueIdentifierService uniqueIdentifierService =
+            new UniqueIdentifierService(IdentifierMappings.Select(mapping => mapping.Value), isLanguageCaseSensitive);
+
+        Func<string, string> customizePropertyName = (name) => { return this.context.EnableNamingAlias ? Customization.CustomizeNaming(name) : name; };
+        var codeDomProvider = (this.context.TargetLanguage == LanguageOption.CSharp ? CSharpProvider : VBProvider);
+        var propertiesToRename = structuredType.DeclaredProperties.Where(prop => !codeDomProvider.IsValidIdentifier(prop.Name)
+            && !LanguageKeywords.Contains(prop.Name));
+
+        foreach (var property in propertiesToRename)
+        {
+            var customizedPropertyName = customizePropertyName(property.Name);
+            var validName = uniqueIdentifierService.GetUniqueIdentifier(GetValidIdentifier(customizedPropertyName));
+            if (IdentifierMappings.ContainsKey(property.Name))
+            {
+                if (!codeDomProvider.IsValidIdentifier(IdentifierMappings[property.Name]))
+                {
+                    IdentifierMappings[property.Name] = validName;
+                }
+            }
+            else
+            {
+                IdentifierMappings.Add(property.Name, validName);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Regex match any non-identifier characters, used to generate valid identifiers<br />
+    /// See C# definitions: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#643-identifiers <br />
+    /// See VB definitions: https://learn.microsoft.com/en-us/dotnet/visual-basic/reference/language-specification/lexical-grammar#identifiers <br />
+    /// See Unicode character class definitions: https://www.unicode.org/reports/tr18/#General_Category_Property
+    /// </summary>
+    private static readonly Regex RegexMatchNonIdentifier = new Regex(@"[^\p{L}\p{Nl}\p{Mn}\p{Mc}\p{Nd}\p{Pc}\p{Cf}]", RegexOptions.Compiled);
+
+    /// <summary>
+    /// Default prefix character used to prefix non-alpha characters in identifiers
+    /// </summary>
+    private const string DefaultPrefixCharacter = "_";
```

Comment:
Since it's only a single character, is it necessary to make it a string?

```suggestion
    private const charDefaultPrefixCharacter = '_';
```